### PR TITLE
feat: download csv from favorites flow

### PR DESCRIPTION
**Pull request by vtsaplin**

### CODE REVIEW

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 1]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
...
+              <RemoveFavoriteIcon />
+              <Text>Delete</Text>
+            </Button>
+            <AlertDialog
+              title="Remove from favorites"
+              variant="confirmation"
+              onPrimaryAction={removeFromFavorites}
+              primaryActionLabel="Confirm"
+              cancelLabel="Cancel">
+              <Text>Are you sure you want to remove the selected variants from your favorites?</Text>
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:151]: Localizable text

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 2]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
...
+              variant="negative"
+              isDisabled={selectedVariants.length === 0}>
+              <RemoveFavoriteIcon />
+              <Text>Delete</Text>
+            </Button>
+            <AlertDialog
+              title="Remove from favorites"
+              variant="confirmation"
+              onPrimaryAction={removeFromFavorites}
+              primaryActionLabel="Confirm"
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:149]: Localizable text

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 3]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
...
+          <DialogTrigger>
+            <Button
+              key="removeFromFavorites"
+              variant="negative"
+              isDisabled={selectedVariants.length === 0}>
+              <RemoveFavoriteIcon />
+              <Text>Delete</Text>
+            </Button>
+            <AlertDialog
+              title="Remove from favorites"
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:146]: Localizable text

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 4]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
...
+            <DeselectAllIcon />
+            <Text>Deselect All</Text>
+          </Button>
+          <DialogTrigger>
+            <Button
+              key="removeFromFavorites"
+              variant="negative"
+              isDisabled={selectedVariants.length === 0}>
+              <RemoveFavoriteIcon />
+              <Text>Delete</Text>
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:143]: Localizable text

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 5]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
...
+            <Text>Select All</Text>
+          </Button>
+          <Button
+            key="deselectAll"
+            variant="secondary"
+            onPress={deselectAll}
+            isSelected={selectedVariants.length === 0}
+            isDisabled={favorites.length === 0}>
+            <DeselectAllIcon />
+            <Text>Deselect All</Text>
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:135]: Localizable text

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 6]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
...
+            <Tooltip>Export the selected variants (text content only) to a CSV file.</Tooltip>
+          </TooltipTrigger>
+          <Button
+            key="selectAll"
+            variant="secondary"
+            onPress={selectAll}
+            isSelected={selectedVariants.length === favorites.length}
+            isDisabled={favorites.length === 0}>
+            <SelectAllIcon />
+            <Text>Select All</Text>
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:126]: Localizable text

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 7]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
...
         </Link>
+        <ButtonGroup marginStart={'auto'}>
+          <TooltipTrigger>
+            <Button
+              key="exportCsv"
+              variant="cta"
+              onPress={exportCsv}
+              isDisabled={selectedVariants.length === 0}>
+              <ExportIcon />
+              <Text>Export to CSV</Text>
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:115]: Localizable text

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 8]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
...
+            </Button>
+            <Tooltip>Export the selected variants (text content only) to a CSV file.</Tooltip>
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:117]: Localizable text

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 9]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
...
+              isDisabled={selectedVariants.length === 0}>
+              <RemoveFavoriteIcon />
+              <Text>Delete</Text>
+            </Button>
+            <AlertDialog
+              title="Remove from favorites"
+              variant="confirmation"
+              onPrimaryAction={removeFromFavorites}
+              primaryActionLabel="Confirm"
+              cancelLabel="Cancel">
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:150]: Localizable text

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 10]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
           <Image src={ChevronLeft} alt={'Back'} width={'24px'} />
           {formatMessage(intlMessages.favoritesView.navigationLabel)}
         </Link>
+        <ButtonGroup marginStart={'auto'}>
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:107]: The right edge of the `Delete` button is not aligned with the variant card's right border.
Screenshot 2024-04-26 at 5 12 30 PM

**Review by askayastha22 — CHANGES_REQUESTED** [review]

File: web-src/src/components/FavoriteVariantListPanel.js [hunk 11]
```diff
@@ -46,6 +104,54 @@ export function FavoriteVariantListPanel(props) {
           <Image src={ChevronLeft} alt={'Back'} width={'24px'} />
```
Comment (askayastha22) [inline web-src/src/components/FavoriteVariantListPanel.js:104]: Localizable alt text

**Comment by coreydulimba-ad** [comment]
@vtsaplin 
image

In the CF Editor the "Generate variations" button when you scroll over it doesn't change to a hand or the button doesn't show a hover state. you click on it and it doesn't look like anything actually happens but then opens a new tab. pleaes make sure the mouse changes state and button changes

**Comment by coreydulimba-ad** [comment]
@vtsaplin the copy icon doesn't seem to work. Steps to reproduce
1. generate copy
2. select copy icon
3. open word file > paste

actual
copy from app doesn't show

expected
copy from app shows

**Comment by coreydulimba-ad** [comment]
@vtsaplin 
generate image button not working
steps to reproduce
1. generate content
2. select generate image

actual
button spins but express window doesn't show

expected
express text to image shows

**Comment by coreydulimba-ad** [comment]
@vtsaplin 
naming issue in export fragment logic
1. generate content
2. select 2nd or 3rd fragment
3. select export fragment

actual
the name from the first variant is pre-filled not the name for the selected variant. appears that the name of first variant always shows in name 
image

showing first name
image

expected
name of selected variant shows

File: web-src/src/components/FavoriteVariantListPanel.js (hunk 10 above)
Comment (vtsaplin) [reply web-src/src/components/FavoriteVariantListPanel.js:107]: From what I see, it should not be aligned, as the left edge of the header also sticks out on the other side. The header is just wider than the cards area.

File: package-lock.json [hunk 12]
```diff
@@ -60,6 +61,10 @@
       },
       "engines": {
         "node": "^14.18 || ^16.13 || >=18"
+      },
+      "optionalDependencies": {
+        "@adobe/adobe-intl-aladdin": "4.3.8-beta.5",
+        "@adobe/adobe-intl-harvester": "3.2.8"
       }
```
Comment (askayastha22) [inline package-lock.json:68]: @vtsaplin Nice addition with optional dependencies. I was just wondering if there is still a possibility that might break CI/CD even with the property of `"optional": true` in private Artifactory packages?

File: web-src/src/components/FavoriteVariantListPanel.js (hunk 11 above)
Comment (askayastha22) [reply web-src/src/components/FavoriteVariantListPanel.js:114]: Nitpicking but I think you missed this one.

**Review by askayastha22 — APPROVED** [review]
LGTM

File: package-lock.json (hunk 12 above)
Comment (vtsaplin) [reply package-lock.json:68]: > I was just wondering if there is still a possibility that might break CI/CD even with the property of "optional": true in private Artifactory packages?

Looks like it's all good—the actions in this PR managed to install dependencies without any issues.

**Review by lenaorobei — APPROVED** [review]

**Pull request merged**